setup_undercloud.yml

```diff
@@ -60,63 +60,93 @@
       set_fact:
         chassis_password:  "{{ instackenv_content.nodes[0].pm_password }}"
 
-    - name: RHEL 8 install for OSP 15 and above
+    - name: Check if /etc/redhat-release exists
+      stat:
+       path: /etc/redhat-release
+      register: rhel_release_stat
+      delegate_to: "{{ undercloud_hostname }}"
+
+    - name: Get RHEL version
+      become: true
+      slurp:
+        src: /etc/redhat-release
+      register: file_content
+      when: rhel_release_stat.stat.exists
+      delegate_to: "{{ undercloud_hostname }}"
+
+    - name: set version fact
+      set_fact:
+        version: "{{ file_content.content | b64decode }}"
+
+    - name: set rhel version
+      set_fact:
+        rhel_version: "{{ version.split()[-2] }}"
+
```

Comment:
We can get this from infrared itself cant we?

Reply:
This is checking for the already installed RHEL version on the host we want to install Undercloud on. I don’t see how we can get it from infrared. Thoughts ?